src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -13,151 +12,39 @@ open System.Threading.Tasks
-            let duration = Stats.LogSink.Restart()
-            if rows > 1 then logActivity "TOTAL" totalCount totalMs
-            let measures: (string * (TimeSpan -> float)) list = [ "s", fun x -> x.TotalSeconds(*; "m", fun x -> x.TotalMinutes; "h", fun x -> x.TotalHours*) ]
-            let logPeriodicRate name count = log.Information("rp{name} {count:n0}", name, count)
-            for uom, f in measures do let d = f duration in if d <> 0. then logPeriodicRate uom (float totalCount/d |> int64)
-
 module private Write =
 
     let private writeEventsAsync (writer: MessageDbWriter) streamName version events ct: Task<MdbSyncResult> =
         writer.WriteMessages(streamName, events, version, ct)
     let inline len (bytes: EventBody) = bytes.Length
     let private eventDataLen (x: IEventData<EventBody>) = len x.Data + len x.Meta
     let private eventDataBytes events = events |> Array.sumBy eventDataLen
-    let private writeEventsLogged writer streamName version events (log: ILogger) ct: Task<MdbSyncResult> = task {
+    let private writeEventsLogged writer streamName version events ct: Task<MdbSyncResult> = task {
         let act = Activity.Current
-        let log = if not (log.IsEnabled Events.LogEventLevel.Debug) then log else log |> Log.propEventData "Json" events
-        let bytes, count = eventDataBytes events, events.Length
-        let log = log |> Log.prop "bytes" bytes
-        if act <> null then act.AddExpectedVersion(version).IncMetric(count, bytes) |> ignore
-        let! t, result = writeEventsAsync writer streamName version events |> Stopwatch.time ct
-        let reqMetric: Log.Measurement = { stream = streamName; interval = t; bytes = bytes; count = count}
-        let resultLog, evt =
-            match result with
-            | MdbSyncResult.Written x ->
-                if act <> null then
-                    act.SetStatus(ActivityStatusCode.Ok).AddTag("eqx.new_version", x) |> ignore
-                log |> Log.prop "currentPosition" x, Log.WriteSuccess reqMetric
-            | MdbSyncResult.ConflictUnknown ->
-                let eventTypes = [| for x in events -> x.EventType |]
-                if act <> null then act.RecordConflict().AddTag("eqx.event_types", eventTypes) |> ignore
-                let writeLog = log |> Log.prop "stream" streamName |> Log.prop "count" count
-                writeLog.Information("MdbTrySync WrongExpectedVersionException writing {eventTypes}, expected {expectedVersion}", eventTypes, version)
-                log, Log.WriteConflict reqMetric
-        (resultLog |> Log.event evt).Information("Mdb{action:l} count={count} conflict={conflict}",
-                                                 "Write", count, match evt with Log.WriteConflict _ -> true | _ -> false)
+        let bytes = eventDataBytes events
+        if act <> null then act.AddExpectedVersion(version).AddAppendBytes(bytes) |> ignore
+        let! result = writeEventsAsync writer streamName version events ct
```

Comment:
I guess there is only one roundtrip and npgsql has a span for it, hm

Reply:
if I'm moving from Cosmos V2 to Cosmos V3 and they change the underlying metric, I'd still like to know my RTT metrics
And if I then move from Cosmos to Dynamo and the SDK does not cover it?
I guess I just need to let go (seealso L72)